runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -36,10 +37,50 @@
 namespace Catalyst::Runtime {
 
 /**
- * @brief Global quantum device unique pointer.
+ * @brief Global quantum device unique pointer with internal linkage.
  */
 thread_local static std::unique_ptr<ExecutionContext> CTX = nullptr;
 
+/**
+ * @brief Thread local device pointer and unique ID with internal linkage.
+ */
+thread_local static QuantumDevice *RTD_PTR = nullptr;
+thread_local static size_t RTD_ID = 0;
```

Comment:
Would it make sense to have the thread_local variable point to the active `RTDevice` instance instead? That way we could eliminate the ID variable (I think).